Fix Edge3 provider navigation with webserver base_url configuration

dheerajturaga opened the pull request:
The Edge3 provider's BrowserRouter was not respecting the Airflow
  webserver.base_url configuration, causing navigation links to generate
  absolute paths from root instead of properly prefixed paths. This
  resulted in 404 errors when Airflow is deployed with a base URL prefix
  (e.g., my_company.com/airflow).

  **Problem:**
  - Edge3 provider pages were navigating to /plugin/edge_worker instead of /airflow/plugin/edge_worker
  - This broke navigation in deployments where nginx redirects traffic from my_company.com/airflow to localhost:8080
  - The issue was in JobsPage.tsx:94 where RouterLink generated links without the base URL prefix

<img width="1642" height="191" alt="image" src="https://github.com/user-attachments/assets/54399bb9-e04a-4c0a-97be-f0916808b23a" />

dheerajturaga committed:
Fix Edge3 provider navigation with webserver base_url configuration

  The Edge3 provider's BrowserRouter was not respecting the Airflow
  webserver.base_url configuration, causing navigation links to generate
  absolute paths from root instead of properly prefixed paths. This
  resulted in 404 errors when Airflow is deployed with a base URL prefix
  (e.g., my_company.com/airflow).

  **Problem:**
  - Edge3 provider pages were navigating to /plugin/edge_worker instead
    of /airflow/plugin/edge_worker
  - This broke navigation in deployments where nginx redirects traffic
    from my_company.com/airflow to localhost:8080
  - The issue was in JobsPage.tsx:94 where RouterLink generated links
    without the base URL prefix

dheerajturaga committed:
Fix: Added _get_base_url_path() helper function and updated all URLs to use it

jscheffl commented on providers/edge3/src/airflow/providers/edge3/plugins/edge_executor_plugin.py:
```diff
@@ -227,6 +227,26 @@ def change_maintenance_comment(self, worker_name: str):
+    if not path.startswith("/"):
+        path = "/" + path
+
+    # Combine paths
```
I assume this is not needed, we have a private method and all references are consistent below.
```suggestion
```

dheerajturaga replied:
I seem to need this, without it im hitting issues

dheerajturaga replied:
<img width="1272" height="692" alt="Image" src="https://github.com/user-attachments/assets/915a1432-4f4d-47ea-b7c1-58851e32a1d6" />

jscheffl replied:
Okay, you are right. But I was mainly referring to the `path` as provided via parameter which is consistent:
```suggestion
    # Normalize paths: remove trailing slash from base, ensure leading slash on path
    base_path = base_path.rstrip("/")
```

jscheffl reviewed (commented):
Can you help me which properties you were setting to make this working with an external nginx? I attempted to fix but got this:

<img width="551" height="218" alt="Image" src="https://github.com/user-attachments/assets/c6555664-98f5-4b32-9fc2-13eb9989f8c2" />

I was setting `export AIRFLOW__API__BASE_URL=/af` and `export AIRFLOW__EDGE__API_URL=http://localhost:8080/af/edge_worker/v1/rpcap` and restarted edge worker and API server.

After this I created a nginx config like
```
    server {
        if ($http_referer ~ ^https?://localhost:9000/af/) {
            rewrite ^ /af$uri;
        }
        location /af/ {
            proxy_pass http://192.168.31.176:28080/;
        }
    }
```
and started a docker with `docker run -p 9000:80 -v $home/Workspace/airflow_review/nginx.conf:/etc/nginx/nginx.conf:ro nginx`

 Was able to log and and browse in http://localhost:9000 but not having the edge plugin calls working. they responded generated requests on nginx with:
172.17.0.1 - - [29/Sep/2025:21:27:35 +0000] "GET /edge_worker/ui/worker HTTP/1.1" 404 22 "http://localhost:9000/af/plugin/edge_worker" "Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/140.0.0.0 Safari/537.36" "-"

dheerajturaga committed:
Fix Edge3 provider frontend API calls with webserver base_url configuration

  The Edge3 provider's frontend was not respecting Airflow's webserver
  base_url configuration, causing API calls to fail when the webserver
  was configured with a subpath (e.g., localhost:8080/airflow).

  The OpenAPI-generated client was using an empty BASE configuration,
  resulting in API calls to /edge_worker/ui/jobs instead of
  /airflow/edge_worker/ui/jobs when base_url was set.

  This fix initializes OpenAPI.BASE from the HTML <base> tag, following
  the same pattern used in Airflow core, ensuring all API calls are
  correctly prefixed with the configured base URL path

jscheffl reviewed (approved):
Looks good. Just like a mini-nit. then I think we can merge.

Sorry took a moment until I had fiddled around making an nginx instance really working

dheerajturaga committed:
Update providers/edge3/src/airflow/providers/edge3/plugins/edge_executor_plugin.py

Co-authored-by: Jens Scheffler <95105677+jscheffl@users.noreply.github.com>

jscheffl reviewed (approved):
Thanks for the fix! Fail in CI of static check is unrelated, therefore merging.
